common/reducers/swap.js

```diff
@@ -0,0 +1,60 @@
+import {
+    SWAP_DESTINATION_AMOUNT,
+    SWAP_DESTINATION_KIND,
+    SWAP_ORIGIN_AMOUNT,
+    SWAP_ORIGIN_KIND,
+    SWAP_UPDATE_BITY_RATES
+} from 'actions/swap';
+
+import {without} from 'lodash';
+const ALL_CRYPTO_KIND_OPTIONS = ['BTC', 'ETH', 'REP'];
+
+const initialState = {
+    originAmount: 0,
+    destinationAmount: 0,
+    originKind: 'BTC',
+    destinationKind: 'ETH',
+    destinationKindOptions: without(ALL_CRYPTO_KIND_OPTIONS, 'BTC'),
+    originKindOptions: without(ALL_CRYPTO_KIND_OPTIONS, 'REP'),
```

Comment:
@NessDan No. you should be able to change origin kind to a currency even if the destination is that already, and the destination should change to the next available option.